### ISSUE TITLE
Updater doesn't respect current location of installation

### ISSUE THREAD

**Issue by scscgit** [nicehash/NiceHashMiner#1855]
If we use a custom location, e.g. `C:\Mining\NiceHash Miner` instead of your default `C:\Users\<user>\AppData\Local\Programs\NiceHash Miner`, the auto-updater ignores that and uses AppData anyway. This has a consequence of breaking our Antivirus exceptions and ignoring the previous configuration (which, by the way, would be nice if you also backed up).

**Comment by luc1an24** [comment]
Hi @scscgit 
Thank you for reaching out.
Because of possible issues caused by custom install paths we decided to disable this feature and force the default one.
If you would like to use custom location you have to download NHM as zip. 
Could I ask you why wouldn't you like to have NHM installed in AppData?
Then you would have to set exception for that path and that is it.
If you will install NHM and then update it the previous configuration will be saved and used in case of NHM version downgrading.

I hope your question/request was answered. If you still have any questions, feel free to open contact us.

**Comment by scscgit** [comment]
Hi, I'm certainly curious about any "possible issues". If you plan to keep this feature disabled, I'd welcome if you could at least print some warning message about the updater ignoring a custom path, such that a zip file is the only viable option (as long as the current instance is run from a custom location, obviously, plus it will save people some time if they mistakenly download the installer, hoping it'll ask for the location, which it doesn't).

Also if there is any security risk of accidentally removing other user's files, I'd personally prefer to have the files moved e.g. to some "old1" sub-directory, or even to a Recycle Bin. There's a minor inconvenience for people using a zip file, such that they have to manually distinguish between configuration and installed files, as simply extracting the zip can leave there some old files that should've been deleted. That's why updater is a most user-friendly approach.

I'm glad that there is finally some config backup, though without auto-updater I haven't experienced it yet - I was mostly concerned because of my old experience with a system crashing (maybe during the loading process), which caused all the configuration to get corrupted and thus broke the auto-start and ruined benchmarks - as that's why I had to make my own scripts with powershell Expand-Archive to re-load the config after boot. I still expect this issue to exist, as there are multiple other apps that suffer the same problem, but it's not related to this issue... :)

About my reason behind using a custom path, firstly I don't consider it acceptable to use a user-specific path under their AppData, as it's not portable on a multi-user setup, or when migrating between computers. Any time you try to use custom scripts, it also becomes problematic that you have to hard-code the username or expand macros like %appdata%. There's also the motivation behind keeping all miners together, and having their path as short a possible. This also keeps antivirus exceptions to a single directory, and possibly even helps with setting up Sandbox whitelist :) But other than that, it's mostly just a subjective preference, and a dislike for apps that abuse AppData how it isn't supposed to be used, though I know Windows may make it hard to modify files outside user directory, as long as the user doesn't set up the access rights :P

**Comment by bpvarsity** [comment]
Let us choose the install path. I do not want it on the main drive

**Comment by bpvarsity** [comment]
Workaround is to extract the nhm_windows_3x.x.x.exe into the directory you want it to be in and then start the app. Also disable auto updates

**Comment by scscgit** [comment]
How many more years will it take before a proper auto-updater gets implemented? I just tried to use it and after it started downloading something, the miner exited without any restart, and of course, after I started it again it still wasn't updated. Shouldn't a double-check to make sure that the miner at least "starts again" be an utmost priority to avoid a downtime? :(

Actually, I even started getting a "NHM experienced too many crashes recently, therefore it will close itself" message instead of NHM launching, and had to delete all files (except config) before re-installing it...
